src/python_ms_core/core/auth/models/permission_request.py

```diff
@@ -0,0 +1,60 @@
+from typing import Optional
+from dataclasses import dataclass
+
+
+@dataclass
+class PermissionRequest:
+    user_id: str
+    org_id: str
+    permissions: Optional[str] = None
+    should_satisfy_all: bool = False
+
+    def __init__(self, user_id: str, org_id: str, permissions: Optional[str] = None, should_satisfy_all: bool = False):
+        self._user_id = user_id
+        self._org_id = org_id
+        self._permissions = permissions
+        self._should_satisfy_all = should_satisfy_all
+
+    @property
+    def user_id(self):
+        return self._user_id
+
+    @user_id.setter
+    def user_id(self, value):
+        self._user_id = value
+
+    @property
+    def org_id(self):
+        return self._org_id
+
+    @org_id.setter
+    def org_id(self, value):
+        self._org_id = value
+
+    @property
+    def permissions(self):
+        return self._permissions
+
+    @permissions.setter
+    def permissions(self, value):
+        self._permissions = value
+
+    @property
+    def should_satisfy_all(self):
+        return self._should_satisfy_all
+
+    @should_satisfy_all.setter
+    def should_satisfy_all(self, value):
+        self._should_satisfy_all = value
+
+    def get_search_params(self):
+        affirmative = 'true' if self._should_satisfy_all else 'false'
+        params = {
+            'userId': self._user_id,
+            'agencyId': self._org_id,
+            'affirmative': affirmative
+        }
+        if self._permissions and len(self._permissions) > 0:
+            params['roles'] = self._permissions
```

Comment:
this may need modification. Check the API URL when there are multiple roles. I think the query formation is a bit different.

Reply:
@susrisha, here we are not forming the query. 
Query Formation happening here - https://github.com/TaskarCenterAtUW/TDEI-python-ms-core/blob/authorizer_implementation/src/python_ms_core/core/auth/provider/hosted/hosted_authorizer.py#L17

Example - 
```python
req = PreparedRequest()
req.prepare_url(url, request_params.get_search_params())

#  url = "https://tdei-auth-n-z-dev.azurewebsites.net/api/v1/hasPermission"

# request_params.get_search_params()   will return 
# {
#  'userId': <USER_ID>,
#  'agencyId': <ORG_ID>,  
#   'affirmative': False,  
#   'roles': ['ROLE_1', `ROLE_2']
# }

response = requests.get(req.url)

# req.url => https://tdei-auth-n-z-dev.azurewebsites.net/api/v1/hasPermission?userId=<USER_ID>&agencyId=<ORG_ID>&affirmative=False&roles=ROLE_1&roles=ROLE_2
return response.json() #=> Ture/False






```